### PR TITLE
bump: Update ESLint packages and migrate to eslint.config.cjs files

### PR DESCRIPTION
**Pull request by JhontSouth**
#minor

## Description
This PR updates the **_ESLint_** packages and applies the respective changes to use _eslint.config.cjs_ files. It also includes the _only-warn_ property to avoid errors with the lint pipeline while the appropriate changes are applied to each library according to the new rules of the latest version of ESLint.

## Specific Changes
  - Updated _**ESLint**_ packages to the latest version.
  - Updated _prettier_ package to the latest version.
  - Removed _.eslintrc.json_ files and added _eslint.config.cjs_ to work with newer versions of _**ESLint**_.
  - Added _only-warn_ property to the main config file.

## Testing
The following image shows the warnings thrown by the _**ESLint**_ scanner.
image

### CODE REVIEW

File: testing/bot-integration/package.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "build": "tsc -b",
     "clean": "rimraf lib tsconfig.tsbuildinfo",
-    "lint": "eslint . --ext .js,.ts",
+    "lint": "eslint .",
```
Comment (ceciliaavila) [inline testing/bot-integration/package.json:9]: This project still has a .eslintrc.json file

File: testing/botbuilder-test-utils/package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
   "scripts": {
     "build": "tsc -b",
     "clean": "rimraf lib tsconfig.tsbuildinfo",
-    "lint": "eslint . --ext .js,.ts"
+    "lint": "eslint ."
```
Comment (ceciliaavila) [inline testing/botbuilder-test-utils/package.json:10]: This still has the .eslintrc.json file.

File: testing/consumer-test/package.json [hunk 3]
```diff
@@ -3,7 +3,7 @@
   "private": true,
   "version": "1.0.0",
   "scripts": {
-    "lint": "eslint . --ext .js,.ts",
+    "lint": "eslint .",
```
Comment (ceciliaavila) [inline testing/consumer-test/package.json:6]: Same, it still has the .eslintrc.json file.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: libraries/adaptive-expressions/package.json [hunk 4]
```diff
@@ -32,6 +37,7 @@
     "big-integer": "^1.6.52",
     "@types/xmldom": "^0.1.34",
     "btoa-lite": "^1.0.0",
+    "eslint-plugin-only-warn": "^1.1.0",
```
Comment (sw-joelmut) [inline libraries/adaptive-expressions/package.json:40]: these should be as devDependencies

File: libraries/adaptive-expressions-ie11/eslint.config.cjs [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+const sharedConfig = require("../../eslint.config.cjs")
+
+module.exports = [
+    js.configs.recommended,
+    ...sharedConfig,
+    {
+        files: ["**/*.js, **/*.ts"],
+    }
+]
```
Comment (sw-joelmut) [inline libraries/adaptive-expressions-ie11/eslint.config.cjs:11]: Example with only-warn
```suggestion
module.exports = [
    ...sharedConfig,
    {
        plugins: {
            "only-warn": onlyWarn,
        }
    }
]
```

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

File: eslint.config.cjs [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+    allConfig: js.configs.all
+});
+
+module.exports = [{
+    ignores: ["**/_ts3.4/", "**/dist/", "**/lib/", "**/node_modules/"],
+}, ...compat.extends(
+    "eslint:recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended",
+    "plugin:jsdoc/recommended",
```
[the remark this answers is not in the document]
Comment (JhontSouth) [reply eslint.config.cjs:34]: No, they have compatibility problems

File: libraries/adaptive-expressions/package.json (hunk 4 above)
Comment (JhontSouth) [reply libraries/adaptive-expressions/package.json:40]: the problem is with depcheck, maybe I could do what you're saying but I would need to skip this libraries in the depcheck analysis

File: libraries/adaptive-expressions-ie11/eslint.config.cjs [hunk 7]
```diff
@@ -0,0 +1,11 @@
+const js = require("@eslint/js");
+
+const sharedConfig = require("../../eslint.config.cjs")
+
+module.exports = [
+    js.configs.recommended,
```
[the remark this answers is not in the document]
Comment (JhontSouth) [reply libraries/adaptive-expressions-ie11/eslint.config.cjs:6]: Okay, I tested and it worked, thx

File: eslint.config.cjs [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+        "mocha/no-return-from-async": ["error"],
+        "mocha/no-sibling-hooks": ["error"],
+        "security/detect-non-literal-fs-filename": "off",
+    },
+}, {
+    files: ["src/**/*.ts"],
+
+    plugins: {
+        jsdoc,
+        "only-warn": onlyWarn,
```
[the remark this answers is not in the document]
Comment (JhontSouth) [reply eslint.config.cjs:167]: fixed

File: testing/bot-integration/package.json (hunk 1 above)
Comment (JhontSouth) [reply testing/bot-integration/package.json:9]: fixed

File: testing/botbuilder-test-utils/package.json (hunk 2 above)
Comment (JhontSouth) [reply testing/botbuilder-test-utils/package.json:10]: fixed

File: libraries/adaptive-expressions/package.json (hunk 4 above)
Comment (JhontSouth) [reply libraries/adaptive-expressions/package.json]: image

Not so good idea

Comment (sw-joelmut) [reply libraries/adaptive-expressions/package.json]: For now, we can disable depcheck for that so it doesnt complain, either way, this is temporary, the only-warn will be removed after we fix each library

File: libraries/botbuilder-azure/package.json [hunk 9]
```diff
@@ -34,10 +34,11 @@
     "bcryptjs": "^2.4.3",
     "botbuilder": "4.1.6",
     "botbuilder-stdlib": "4.1.6",
+    "eslint-plugin-only-warn": "^1.1.0",
     "lodash": "^4.17.20",
     "p-map": "^4.0.0"
   },
-  "devDependencies": {
+  "devDependencies": {
```
Comment (ceciliaavila) [inline libraries/botbuilder-azure/package.json:41]: There's an extra space here.

File: libraries/botbuilder-core/package.json [hunk 10]
```diff
@@ -31,11 +31,12 @@
     "botbuilder-stdlib": "4.1.6",
     "botframework-connector": "4.1.6",
     "botframework-schema": "4.1.6",
+    "eslint-plugin-only-warn": "^1.1.0",
     "uuid": "^10.0.0",
     "zod": "^3.23.8"
   },
   "devDependencies": {
-    "axios": "^1.7.7",
+    "axios": "^1.7.7",
```
Comment (ceciliaavila) [inline libraries/botbuilder-core/package.json:39]: extra space here

File: libraries/botbuilder-dialogs-adaptive-testing/package.json [hunk 11]
```diff
@@ -32,13 +32,14 @@
...
     "botbuilder-stdlib": "4.1.6",
+    "eslint-plugin-only-warn": "^1.1.0",
     "murmurhash-js": "^1.0.0",
     "nock": "^13.5.5",
     "url-parse": "^1.5.10",
     "zod": "^3.23.8"
   },
   "devDependencies": {
-    "botbuilder": "4.1.6",
+    "botbuilder": "4.1.6",
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs-adaptive-testing/package.json:42]: extra space here

File: libraries/botbuilder-dialogs/package.json [hunk 12]
```diff
@@ -34,13 +34,15 @@
...
     "botframework-connector": "4.1.6",
+    "eslint-plugin-only-warn": "^1.1.0",    
     "globalize": "^1.7.0",
     "lodash": "^4.17.21",
     "uuid": "^10.0.0",
     "zod": "^3.23.8"
   },
   "devDependencies": {
     "@types/globalize": "^1.5.5",
+
```
Comment (ceciliaavila) [inline libraries/botbuilder-dialogs/package.json:45]: Is this line necessary?

File: libraries/botbuilder/package.json [hunk 13]
```diff
@@ -36,14 +36,15 @@
...
+    "eslint-plugin-only-warn": "^1.1.0",
     "filenamify": "^4.3.0",
     "fs-extra": "^11.2.0",
     "htmlparser2": "^9.0.1",
     "uuid": "^10.0.0",
     "zod": "^3.23.8"
   },
   "devDependencies": {
-    "chai": "^4.5.0",
+    "chai": "^4.5.0",
```
Comment (ceciliaavila) [inline libraries/botbuilder/package.json:47]: There's a space here.

File: libraries/botframework-config/package.json [hunk 14]
```diff
@@ -34,17 +34,18 @@
     "url": "git@github.com:Microsoft/botbuilder-js.git"
   },
   "dependencies": {
+    "eslint-plugin-only-warn": "^1.1.0",
     "fs-extra": "^11.2.0",
     "uuid": "^10.0.0"
   },
   "devDependencies": {
-    "@types/uuid": "^10.0.0"
+    "@types/uuid": "^10.0.0"
```
Comment (ceciliaavila) [inline libraries/botframework-config/package.json:42]: Space here

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Review by sw-joelmut — APPROVED** [review]

**Comment by JhontSouth** [comment]
Promoted /4776

**Pull request closed without merging**